### PR TITLE
Fixes #23979 - add option to search location & organization

### PR DESCRIPTION
**Pull request by shiramax**

### CODE REVIEW

File: app/controllers/api/v2/fact_values_controller.rb [hunk 1]
```diff
@@ -18,6 +18,10 @@ def index
         host_id = params[:host_id]
         @fact_values = FactValue.build_facts_hash(values, host_id)
       end
+
+      def setup_search_options
+        params[:search] = " host = " + params[:host_id] if params[:host_id]
```
Comment (ohadlevy) [inline app/controllers/api/v2/fact_values_controller.rb]: can you please add a test for this? thanks

Comment (tbrisker) [inline app/controllers/api/v2/fact_values_controller.rb]: this should be added to the search param, not override it. also keep in mind the param might not be there.

**Review by tbrisker — CHANGES_REQUESTED** [review]

**Comment by shiramax** [comment]
thanks @tbrisker @ohadlevy , please review again :)

File: app/controllers/api/v2/fact_values_controller.rb [hunk 2]
```diff
@@ -18,6 +18,11 @@ def index
         host_id = params[:host_id]
         @fact_values = FactValue.build_facts_hash(values, host_id)
       end
+
+      def setup_search_options
+        params[:search] ||= ""
+        params[:search] += " host = " + params[:host_id] if params[:host_id]
```
Comment (adamruzicka) [inline app/controllers/api/v2/fact_values_controller.rb:24]: Doesn't the original `Api::BaseController#setup_search_options`[1] do exactly this, but in a slightly more generic way?

[1] - https://github.com/theforeman/foreman/blob/develop/app/controllers/api/base_controller.rb#L238

Comment (shiramax) [reply app/controllers/api/v2/fact_values_controller.rb:24]: yes, the problem is that the original  setup_search_options functions apply also to the location and organization search too, which preventing the api/cli to search by location or organization.

Comment (adamruzicka) [reply app/controllers/api/v2/fact_values_controller.rb:24]: Ah, I see

**Comment by adamruzicka** [comment]
Looks good to me, @tbrisker, @ohadlevy anything to change here?

**Review by tbrisker — APPROVED** [review]
Thanks @shiramax !

**Pull request merged**